SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,412 @@
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "Tests/AutomationCommon.h"
+#include "Tests/AutomationEditorCommon.h"
+#include "Tests/TestDefinitions.h"
+#include "Engine/Engine.h"
+#include "GameFramework/GameStateBase.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include <vector>
+#include <utility>
+#include "SpatialGDKTests/SpatialGDK/LoadBalancing/GridBasedLBStrategy/TestGridBasedLBStrategy.h"
+#include <string>
+#include "TestLayeredLBStrategy.h"
+#include "GameFramework/DefaultPawn.h"
+#include "Misc/Optional.h"
+
+
+#define LAYEREDLBSTRATEGY_TEST(TestName) \
+	GDK_TEST(Core, ULayeredLBStrategy, TestName)
+
+
+
+namespace
+{
+	ULayeredLBStrategy* Strat;
+	UWorld* TestWorld;
+	TMap<FName, AActor*> TestActors;
+
+	// Copied from AutomationCommon::GetAnyGameWorld().
+	UWorld* GetAnyGameWorld()
+	{
+		UWorld* World = nullptr;
+		const TIndirectArray<FWorldContext>& WorldContexts = GEngine->GetWorldContexts();
+		for (const FWorldContext& Context : WorldContexts)
+		{
+			if ((Context.WorldType == EWorldType::PIE || Context.WorldType == EWorldType::Game)
+				&& (Context.World() != nullptr))
+			{
+				World = Context.World();
+				break;
+			}
+		}
+
+		return World;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND(FWaitForWorld);
+	bool FWaitForWorld::Update()
+	{
+		TestWorld = GetAnyGameWorld();
+
+		if (TestWorld && TestWorld->AreActorsInitialized())
+		{
+			AGameStateBase* GameState = TestWorld->GetGameState();
+			if (GameState && GameState->HasMatchStarted())
+			{
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	struct StrategyConfig {
+		TSubclassOf<UAbstractLBStrategy> StrategyClass;
+		TSet<TSoftClassPtr<AActor>> TargetActorTypes;
+	};
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FSetupStrategy,
```

Comment:
In general, we don't want to use std::vector, but I also think this function is doing a bit too much.
Maybe instead of passing in the vector of Strategies, have one command which takes one strategy and appends it to the WorkerLayers, and then the tests can call that multiple times, once for each layer. That would also give you the option to have a flag for whether it appends to the layers or sets a Strategy as the default.
Then this could just call Init, and use FSetupStrategysLocalWorker to set the local worker.
It makes the test cases a bit longer, but I think also easier to read.

Reply:
done